### DISCUSSION TITLE
Why can you not invoke reset  inside onSubmit

### DISCUSSION THREAD

**Discussion by brandonwisnicki** [react-hook-form/react-hook-form#6078] [Q&A] [answered]
Per V7's rules for `reset`, you should not "invoke `reset` inside onReset or onSubmit callback." However, it seems to work fine with my code. Why can we not invoke reset there, and why must we invoke it via a useEffect hook? Thanks!

**Comment by bluebill1049** [comment, marked as answer]
- Because onSubmit is async include its validation and reset inside the callback will intercept the result of formState update.
- We can't set value for native input during onReset event.

**Reply by bluebill1049** [reply]
I think your use case is fine because you are not subscribed to the `formState`

**Reply by brandonwisnicki** [reply]
Got it, so as long as you avoid the formState in onSubmit, it should be okay. Thanks so much!
